code/modules/food_and_drinks/food.dm

```diff
@@ -15,6 +15,7 @@
 	var/foodtype = NONE
 	var/last_check_time
 	var/in_container = FALSE //currently just stops "was bitten X times!" messages on canned food
+	var/wine_power = 10 //Determines the boozepwr of the ale.
```

Comment:
See above - componentization would cut down on the number of variables added to various locations, and allow you to give non-foods winepower.

Reply:
Since GetComponent isn't allowed I don't know how to feasibly do this.

Reply:
A component isn't the answer here but a signal can do this nicely in a generic way.

Signals are best applied when there's a single source and it describes the event happening rather than what should be the result of the signal call. You can do both here by replacing the call to the on_brew proc with a signal something like this

```
var/list/brew_results = list()
var/brew_flags = SEND_SIGNAL(thing, COMSIG_ATOM_ON_BREW, brew_results)
...dostuff
if(flags & COMPONENT_BREW_CONSUME)
  qdel(thing)
```

Lists passed in with signals and edited by the receiver also get edited outside the signal so you can use the changes elsewhere. This in a generic way allows for things to specify what results they want from brewing. Note that I had the signal also return a flag to qdel so that by default nothing happens. You'll probably also want nothing to happen if no flags and no brew results are given back out.

Reply:
resolved, we're using 2020 signals now, goofcode in the future :tm: